src/toolchains.ts

```diff
@@ -0,0 +1,85 @@
+import * as path from 'path';
+import * as core from '@actions/core';
+import * as io from '@actions/io';
+
+import * as fs from 'fs';
+import * as os from 'os';
+
+import { create as xmlCreate } from 'xmlbuilder2';
+import * as constants from './constants';
+import { M2_DIR, MVN_TOOLCHAINS_FILE } from './constants';
+import { getBooleanInput } from './util';
+
+export async function configureToolchains(jdkHome: string) {
+  const version = core.getInput(constants.INPUT_JAVA_VERSION);
+  const vendor = core.getInput(constants.INPUT_DISTRIBUTION);
+  const settingsDirectory =
+    core.getInput(constants.INPUT_SETTINGS_PATH) || path.join(os.homedir(), M2_DIR);
+  const overwriteSettings = getBooleanInput(constants.INPUT_OVERWRITE_SETTINGS, true);
+
+  await createToolchainsSettings(version, vendor, jdkHome, settingsDirectory, overwriteSettings);
+}
+
+export async function createToolchainsSettings(
+  version: string,
+  vendor: string,
+  jdkHome: string,
+  settingsDirectory: string,
+  overwriteSettings: boolean
+) {
+  core.info(`Creating ${MVN_TOOLCHAINS_FILE} for JDK version ${version} from ${vendor}`);
+  // when an alternate m2 location is specified use only that location (no .m2 directory)
+  // otherwise use the home/.m2/ path
+  await io.mkdirP(settingsDirectory);
+  await write(settingsDirectory, generate(version, vendor, jdkHome), overwriteSettings);
+}
+
+// only exported for testing purposes
+export function generate(version: string, vendor: string, jdkHome: string) {
+  const xmlObj: { [key: string]: any } = {
+    toolchains: {
+      '@xmlns': 'http://maven.apache.org/TOOLCHAINS/1.1.0',
+      '@xmlns:xsi': 'http://www.w3.org/2001/XMLSchema-instance',
+      '@xsi:schemaLocation':
+        'http://maven.apache.org/TOOLCHAINS/1.1.0 http://maven.apache.org/xsd/toolchains-1.1.0.xsd',
+      toolchain: [
+        {
+          type: 'jdk',
```

Comment:
Maybe it would be possible to have an "id" ([like described here](https://maven.apache.org/plugins/maven-toolchains-plugin/toolchains/jdk.html)) defined in the action as a parameter that is then used here? like
`id: '<id from action>'` t

Reply:
@laeubi that should be possible now

Reply:
Man thanks, lets hope this get merged soon!